instrumentation/aws-sdk/aws-sdk-2.2/testing/src/main/groovy/io/opentelemetry/instrumentation/awssdk/v2_2/AbstractAws2ClientTest.groovy

```diff
@@ -120,12 +132,92 @@ abstract class AbstractAws2ClientTest extends InstrumentationSpecification {
 
     expect:
     response != null
-    assertDynamoDbRequest(service, operation, path, method, requestId)
+    switch (operation) {
+      case "CreateTable":
+        assertCreateTableRequest(path, method, requestId)
+        break
+      case "Query":
+        assertQueryRequest(path, method, requestId)
+        break
+      default:
+        assertDynamoDbRequest(service, operation, path, method, requestId)
+    }
 
     where:
     [service, operation, method, path, requestId, builder, call] << dynamoDbRequestDataTable(DynamoDbAsyncClient.builder())
   }
 
+  def assertCreateTableRequest(path, method, requestId) {
+    assertTraces(1) {
+      trace(0, 1) {
+        span(0) {
+          name "DynamoDb.CreateTable"
+          kind CLIENT
+          errored false
+          hasNoParent()
+          attributes {
+            "${SemanticAttributes.NET_TRANSPORT.key}" "IP.TCP"
+            "${SemanticAttributes.NET_PEER_NAME.key}" "localhost"
+            "${SemanticAttributes.NET_PEER_PORT.key}" server.address.port
+            "${SemanticAttributes.HTTP_URL.key}" { it.startsWith("${server.address}${path}") }
+            "${SemanticAttributes.HTTP_METHOD.key}" "$method"
+            "${SemanticAttributes.HTTP_STATUS_CODE.key}" 200
+            "${SemanticAttributes.HTTP_USER_AGENT.key}" { it.startsWith("aws-sdk-java/") }
+            "${SemanticAttributes.HTTP_FLAVOR.key}" "1.1"
+            "aws.service" "DynamoDb"
+            "aws.operation" "CreateTable"
+            "aws.agent" "java-aws-sdk"
+            "aws.requestId" "$requestId"
+            "aws.table.name" "sometable"
+            "${SemanticAttributes.DB_SYSTEM.key}" "dynamodb"
+            "${SemanticAttributes.DB_NAME.key}" "sometable"
+            "${SemanticAttributes.DB_OPERATION.key}" "CreateTable"
+            "aws.dynamodb.global_secondary_indexes" "{\"IndexName\":\"globalIndex\",\"KeySchema\":[{\"AttributeName\":\"attribute\"}],\"ProvisionedThroughput\":{\"ReadCapacityUnits\":10,\"WriteCapacityUnits\":12}}"
```

Comment:
`global_secondary_indexes` is a list field (I think), it looks like this is doing a single value assertion?

Reply:
Serializer was implemented to just join elements with a delimiter, without parentheses denoting an array. Changed that and incidentally discovered that AWS SDK marshaller can't be re-used (throws NPE in such case!). So thanks for pointing that out :)